zipkin-storage/elasticsearch/src/main/java/zipkin/storage/elasticsearch/ElasticsearchSpanConsumer.java

```diff
@@ -42,24 +44,35 @@
 
   @Override
   public ListenableFuture<Void> accept(List<Span> spans) {
-    BulkRequestBuilder request = client.prepareBulk();
-    for (Span span : spans) {
-      request.add(createSpanIndexRequest(ApplyTimestampAndDuration.apply(span)));
+    if (spans.isEmpty()) return Futures.immediateFuture(null);
+
+    // Create a bulk request when there is more than one span to store
+    ListenableFuture<?> future;
```

Comment:
sadly can't unless we refactor further due to the later if
(ElasticsearchStorage.FLUSH_ON_WRITES)